### DISCUSSION TITLE
Locked out of Database

### DISCUSSION THREAD

**Discussion by B08Z** [babybuddy/babybuddy#712] [Support]
We have been using baby buddy for 3 months with our triplets it's been such a useful tool. However, for some reason I had to log back in and unfortunately managed to lock my self out of the database. 

I have seen that I can use manage.py but it's not available in the docker image. 

Is there any advice on how to unlock the dB when running BB in a docker.

**Comment by cdubz** [comment]
The management command should be accessible even in Docker -- it can just be a little tough to get working...

Does this discussion help at all? https://github.com/babybuddy/babybuddy/issues/676#issuecomment-1656726451

**Comment by barakplasma** [comment]
It's likely that your db is an SQLite file in the docker container, or in a docker volume. You might be able to `docker cp` to back it up.

I setup litestream » B2 storage for my BB instance for backup purposes.

**Comment by B08Z** [comment]
> It's likely that your db is an SQLite file in the docker container, or in a docker volume. You might be able to `docker cp` to back it up.
> 
> I setup litestream » B2 storage for my BB instance for backup purposes. 

I'll have a look at doing this but I thought the DB was locked out even if I found it.

**Comment by B08Z** [comment]
> The management command should be accessible even in Docker -- it can just be a little tough to get working...
> 
> Does this discussion help at all? https://github.com/babybuddy/babybuddy/issues/676#issuecomment-1656726451

Hmm, I'm not sure it does. You say the mange.py command should be available in docker could you explain how I might run it?

**Comment by cdubz** [comment]
> Hmm, I'm not sure it does. You say the mange.py command should be available in docker could you explain how I might run it?

Docker is not an area where I am particularly strong but I think you should be able to shell in to the container and use it, right?

**Comment by B08Z** [comment]
> > Hmm, I'm not sure it does. You say the mange.py command should be available in docker could you explain how I might run it?
> 
> Docker is not an area where I am particularly strong but I think you should be able to shell in to the container and use it, right?

I can shell into the docker container but I have no idea where to run manage.py from. It doesn't seem to exist within the container. 

Also I'd happily copy the DB out but can't seem to find where it's stored.

**Comment by cdubz** [comment]
Here is the old issue comment I was trying to find -- https://github.com/babybuddy/babybuddy/issues/273#issuecomment-894529756

Does that help? If so we should also document this... (:

**Comment by nitsujesor** [comment]
Hi @cdubz, @B08Z,

I had the same issue and was able to resolve it with the following steps:
1. enter the Baby Buddy docker shell: 
```
docker exec -it babybuddy sh
```
2. set Django/BB settings: 
```
export \
    DJANGO_SETTINGS_MODULE="babybuddy.settings.base" \
    ALLOWED_HOSTS="${ALLOWED_HOSTS:-*}" \
    TIME_ZONE="${TZ:-UTC}" \
    DEBUG="${DEBUG:-False}" \
    SECRET_KEY="${SECRET_KEY:-$(cat /config/.secretkey)}"
```
3. Create new "recovery" user: 
```
python /app/www/public/manage.py createuser --username recovery --password recovery --is-staff
```
4. Login to recovery user, reset login information as required.

Thanks to the efforts of the users in #273 and #676 for laying out the individual steps!

**Comment by B08Z** [comment]
> Hi @cdubz, @B08Z,
> 
> I had the same issue and was able to resolve it with the following steps:
> 1. enter the Baby Buddy docker shell: `docker exec -it babybuddy sh`
> 2. set Django/BB settings: `export \
>     DJANGO_SETTINGS_MODULE="babybuddy.settings.base" \
>     ALLOWED_HOSTS="${ALLOWED_HOSTS:-*}" \
>     TIME_ZONE="${TZ:-UTC}" \
>     DEBUG="${DEBUG:-False}" \
>     SECRET_KEY="${SECRET_KEY:-$(cat /config/.secretkey)}"`
> 3. Create new "recovery" user: `python /app/www/public/manage.py createuser --username recovery --password recovery --is-staff`
> 4. Login to recovery user, reset login information as required.
> 
> Thanks to the efforts of the users in #273 and #676 for laying out the individual steps!

Thanks so much.

I just tried but get an error:

>: export \ DJANGO_SETTINGS_MODULE="babybuddy.settings.base" \ ALLOWED_HOSTS="${ALLOWED_HOSTS:-*}" \ TIME_ZONE="${TZ:-UTC}" \ DEBUG="${DEBUG:-False}" \ SECRET_KEY="${SECRET_KEY:-$(cat /config/.secretkey)}"

> sh: export:  DJANGO_SETTINGS_MODULE: bad variable name


Sorry for poor formatting on a phone trying to feed a baby;)

**Comment by nitsujesor** [comment]
Sorry, my previous formatting was imprecise. I was able to replicate your error when copy/pasting what I had above in Step 2. Please copy/paste the block as it is in #676. I will edit my above comment to reflect this change.

```
export \
    DJANGO_SETTINGS_MODULE="babybuddy.settings.base" \
    ALLOWED_HOSTS="${ALLOWED_HOSTS:-*}" \
    TIME_ZONE="${TZ:-UTC}" \
    DEBUG="${DEBUG:-False}" \
    SECRET_KEY="${SECRET_KEY:-$(cat /config/.secretkey)}"
```

**Comment by B08Z** [comment]
So,

Once again thanks so much I am now onto the app and can view users in the database. 

I have changed the password and logged back in under that user but all the data has gone. 

Is this expected behaviour? I fear I might have messed up the DB when trying to get back in. 

In my docker config folder I can see the sqlight DB and see that it's large as in looks like it has data? 

I hope I'm getting closer. 

Thanks again this app is a complete life saver with multiples.

**Comment by MrApplejuice** [comment]
You should be able to tell if there is any data in the database by clicking on any of the activities:

![image](https://github.com/babybuddy/babybuddy/assets/542105/0ab26ad0-4f60-4a09-96e1-57a6f3d64625)

This is a screenshot of the sleep entries in of by Babybuddy test instance. It should be relatively straightforward to read.

**Comment by B08Z** [comment]
Thanks for coming back to me. 

Attached are the screen shots from the DB admin page and also one when I run ls -la of the DB file. 

I can see that the db.sqlite file has data in but as you can see from the screen shots there is nothing apearing in the app. 

![Screenshot 2023-10-04 at 08 16 48](https://github.com/babybuddy/babybuddy/assets/19637029/7ae9c03d-c31e-4e90-8103-41c6f50521df)
![Screenshot 2023-10-04 at 08 14 30](https://github.com/babybuddy/babybuddy/assets/19637029/4e563b4c-3e10-4bb2-97c8-023fed60cada)
![Screenshot 2023-10-04 at 08 14 19](https://github.com/babybuddy/babybuddy/assets/19637029/1a151bbd-0425-4495-a59c-20bc5850f744)

**Comment by MrApplejuice** [comment]
Yeah, that unfortunately looks very very empty, indeed. You could try to download the database file and view it in https://sqlitebrowser.org/ . Maybe that there is something left in there.
